pkg/apiserver/registry/networkpolicy/ipassociation/rest_test.go

```diff
@@ -0,0 +1,322 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipassociation
+
+import (
+	"net"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+
+	"antrea.io/antrea/pkg/apis/controlplane"
+	"antrea.io/antrea/pkg/apis/crd/v1alpha2"
+	fakeversioned "antrea.io/antrea/pkg/client/clientset/versioned/fake"
+	crdinformers "antrea.io/antrea/pkg/client/informers/externalversions"
+	"antrea.io/antrea/pkg/controller/grouping"
+	"antrea.io/antrea/pkg/controller/types"
+	"antrea.io/antrea/pkg/util/k8s"
+)
+
+const informerDefaultResync = 30 * time.Second
+
+type fakeIPBQuerier struct {
+	ipGroupMap map[string][]types.Group
+}
+
+func (iq fakeIPBQuerier) GetAssociatedIPBlockGroups(ip net.IP) ([]types.Group, error) {
+	if groupList, ok := iq.ipGroupMap[ip.String()]; ok {
+		return groupList, nil
+	}
+	return nil, nil
+}
+
+type fakeQuerier struct {
+	groups map[string][]types.Group
+}
+
+func (q fakeQuerier) GetAssociatedGroups(name, namespace string) ([]types.Group, error) {
+	memberKey := k8s.NamespacedName(namespace, name)
+	if refs, ok := q.groups[memberKey]; ok {
+		return refs, nil
+	}
+	return []types.Group{}, nil
+}
+
+func TestREST(t *testing.T) {
+	r := NewREST(nil, nil, nil, nil)
+	assert.Equal(t, &controlplane.IPGroupAssociation{}, r.New())
+	assert.False(t, r.NamespaceScoped())
+}
+
+func TestRESTGet(t *testing.T) {
```

Comment:
It will be better to include a nested group test case.

Reply:
The group querier in this testcase is faked; it relies on the correct result of `GetAssociatedGroups`, which has UT on its own. Nested groups are comprehensively tested in that pkg.